src/components/PaletteMRC.ts

```diff
@@ -84,41 +85,93 @@ export class PaletteMRC extends MarkdownRenderChild {
             // Extract JSON settings from the palette
             return JSON.parse(input);
         }
-        catch(error) {
+        catch (error) {
             return Status.INVALID_SETTINGS;
         }
     }
 
+    /**
+     * Extracts the color value from the frontmatter
+     * @param frontMatter from the linked file
+     * @returns string from the property or null if it does not exist
+     */
+    private parseColorFromFrontMatter(frontMatter: FrontMatterCache): string | null {
+        if (frontMatter![this.pluginSettings.propertyKeyColor]) {
+            return frontMatter[this.pluginSettings.propertyKeyColor];
+        }
+        return null;
+    }
+
+    /**
+ * Extracts the alias override value from the frontmatter
+ * @param frontMatter from the linked file
+ * @returns string from the property or null if it does not exist
+ */
+    private parseNameFromFrontMatter(frontMatter: FrontMatterCache): string | null {
+        return frontMatter[this.pluginSettings.propertyKeyAlias] ?? null;
+    }
+
+    /**
+     * Converts all indiret references (internal links & URLs) into direct colors.
+     * @param colors input color array strings
+     * @param settings 
+     * @returns string[] of direct colors or status error
+     */
+    private extractColors(colors: string[], settings: PaletteSettings): string[] | Status {
+        //const vault = this.plugin.app.vault;
+        for (let i = 0; i < colors.length; i++) {
+
+            // Convert URLs to colors
+            if (colors[i].match(urlRegex)) {
+                const parsedColors = parseUrl(colors[i]);
+                colors.splice(i, 1, ...parsedColors);
+            }
+
+            // Convert internal links to colors
+            if (this.wikiLinkRegex.test(colors[i])) {
+                let linkPath = getLinkpath(colors[i]);
+                linkPath = linkPath.substring(2, linkPath.length - 2);

```

Comment:
This doesn't seem like it will work with renamed internal links, like from your example below [[Example#Details|Section name]]. It might be appropriate to take the text before any of the delimiting characters (like `|` or `#`)